doc/advanced-testing-mini-course/src/Lec04FaultInjection.lhs

```diff
@@ -88,9 +99,32 @@ Faulty queue
 >   res2 <- qiEnqueue (ffqQueue ffq) "test2"
 >   assert (res2 == False) (return ())
 
+
+-- XXX: make `exec` :: Command -> Model -> IO {Fail, Info, Ok Response}
+
+-- XXX: stop testing if sequential property hits Info, otherwise we break:
+
+--     Crashes
+--
+--     If an operation does not complete for some reason (perhaps because it
+--     timed out or a critical component crashed) that operation has no
+--     completion time, and must, in general, be considered concurrent with
+--     every operation after its invocation. It may or may not execute.
+--
+--     A process with an operation is in this state is effectively stuck, and
+--     can never invoke another operation again. If it were to invoke another
+--     operation, it would violate our single-threaded constraint: processes
+--     only do one thing at a time.
+--
+-- Source: https://jepsen.io/consistency
+
 Discussion
 ----------
 
+- Modelling the faults, i.e. move non-determinism out from linearisability
```

Comment:
```suggestion
- Modelling the faults, i.e. move some non-determinism out from linearisability
```
There is still a lot of non-determinism in the concurrent case even without faults.